File: app/src/main/java/com/zskjprojectj/andouclient/activity/QrCodeActivity.java
package com.zskjprojectj.andouclient.activity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.SurfaceView;
import android.view.View;

import com.blankj.utilcode.util.ToastUtils;
import com.king.zxing.CaptureHelper;
import com.king.zxing.ViewfinderView;
import com.zskjprojectj.andouclient.R;
import com.zskjprojectj.andouclient.base.BaseActivity;
import com.zskjprojectj.andouclient.base.BasePresenter;
import com.zskjprojectj.andouclient.http.ApiUtils;
import com.zskjprojectj.andouclient.http.BaseObserver;
import com.zskjprojectj.andouclient.http.HttpRxObservable;
import com.zskjprojectj.andouclient.utils.LoginInfoUtil;
import com.zskjprojectj.andouclient.utils.SharedPreferencesManager;

import java.io.IOException;

import butterknife.BindView;
import mlxy.utils.L;

public class QrCodeActivity extends BaseActivity {

    CaptureHelper mCaptureHelper;
    private String erweicode;

    @Override
    protected void setRootView() {
        setContentView(R.layout.activity_qr_code);
    }

    @Override
    protected void initViews() {
        setTitle("扫一扫");
        SurfaceView surfaceView = findViewById(R.id.surfaceView);
        ViewfinderView viewfinderView = findViewById(R.id.viewfinderView);
        View ivTorch = findViewById(R.id.ivTorch);
        mCaptureHelper = new CaptureHelper(this, surfaceView, viewfinderView, ivTorch);
        mCaptureHelper.setOnCaptureCallback(result -> {
            //ToastUtils.showShort(result);
            HttpRxObservable.getObservable(ApiUtils.getApiService().binding(LoginInfoUtil.getUid(),LoginInfoUtil.getToken(),result)).subscribe(new BaseObserver<Object>(mAt) {
                @Override
                public void onHandleSuccess(Object o) throws IOException {

                }
            });

            return true;
        });
        mCaptureHelper.onCreate();
        mCaptureHelper.vibrate(true)
                .fullScreenScan(true)//全屏扫码
                .continuousScan(false);

    }

    @Override
    protected void initData(Bundle savedInstanceState) {

    }

    @Override
    public void getDataFromServer() {

    }

    @Override
    protected BasePresenter createPresenter() {
        return null;
    }


    @Override
    protected void onResume() {
        super.onResume();
        mCaptureHelper.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mCaptureHelper.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mCaptureHelper.onDestroy();
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        mCaptureHelper.onTouchEvent(event);
        return super.onTouchEvent(event);
    }
}

File: app/src/main/java/com/zskjprojectj/andouclient/activity/QrCodeActivity.kt
package com.zskjprojectj.andouclient.activity

import android.os.Bundle
import android.view.MotionEvent
import android.view.SurfaceView
import android.view.View
import com.blankj.utilcode.util.ToastUtils
import com.king.zxing.CaptureHelper
import com.king.zxing.ViewfinderView
import com.zhuosongkj.android.library.app.BaseActivity
import com.zhuosongkj.android.library.util.ActionBarUtil
import com.zhuosongkj.android.library.util.RequestUtil
import com.zskjprojectj.andouclient.R
import com.zskjprojectj.andouclient.http.ApiUtils
import com.zskjprojectj.andouclient.utils.LoginInfoUtil


class QrCodeActivity : BaseActivity() {
    var mCaptureHelper: CaptureHelper? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActionBarUtil.setTitle(mActivity, "扫一扫")
        val surfaceView = findViewById<SurfaceView>(R.id.surfaceView)
        val viewfinderView: ViewfinderView = findViewById(R.id.viewfinderView)
        val ivTorch = findViewById<View>(R.id.ivTorch)
        mCaptureHelper = CaptureHelper(this, surfaceView, viewfinderView, ivTorch)
        mCaptureHelper?.setOnCaptureCallback { result: String ->
            RequestUtil.request(mActivity, true, false, {
                ApiUtils.getApiService().binding(LoginInfoUtil.getUid(), LoginInfoUtil.getToken(), result)
            }, {
                ToastUtils.showShort(it.msg)
                finish()
            })
            true
        }
        mCaptureHelper?.onCreate()
        mCaptureHelper?.vibrate(true)
                ?.fullScreenScan(true) //全屏扫码
                ?.continuousScan(false)
    }

    override fun onResume() {
        super.onResume()
        mCaptureHelper?.onResume()
    }

    override fun onPause() {
        super.onPause()
        mCaptureHelper?.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
        mCaptureHelper?.onDestroy()
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        mCaptureHelper?.onTouchEvent(event)
        return super.onTouchEvent(event)
    }

    override fun getContentView(): Int {
        return R.layout.activity_qr_code
    }
}